cmd/mrcalc.cpp

```diff
@@ -480,13 +480,12 @@ class LoadedImage {
 
 class StackEntry {
 public:
-  StackEntry(const char *entry) : arg(entry), rng_gaussian(false), image_is_complex(false) {}
+  StackEntry(std::string entry) : arg(std::move(entry)), rng_gaussian(false), image_is_complex(false) {}
```

Comment:
Is the `std::move()` here necessary? If yes it could do with code comment to explain why.

Reply:
Not strictly required, but it's a performance optimisation and conveys the intent better I think. `StackEntry` needs to store `entry` in `arg` which requires a copy if a non-temporary value is passed to the constructor. However, the copy is unnecessary if a rvalue (temporary) is passed, which is what the `std::move` avoids. The alternative would be to pass by const ref, but then copies would occur in both cases (unless an additional && overload is provided). See [here](https://stackoverflow.com/a/51706522).